heroku/cli
# refactor(redis-v5): Move command 'redis:maintenance' to CLI

## Description

Moves the `redis:maintenance` command to CLI, converting to OClif Core and TS.

## Test instructions

You can test by pulling this branch, installing dependencies with `yarn` and building with `yarn build`. Then run the following commands, knowing that if the app doesn't have any Redis add-on, you should see an error and in apps with more than one Redis add-on you should see an error unless you specify the DATABASE arg.

```
./bin/run redis:maintenance --help                                                 # always test that help looks ok.
./bin/run redis:maintenance -a test_app_without_redis                              # shows error.
./bin/run redis:maintenance -a test_app_single_redis                               # shows current weekly maintenance window.
./bin/run redis:maintenance -a test_app_single_redis -w "Tue 23:00"                # sets the weekly maintenance window.
./bin/run redis:maintenance -a test_app_single_redis --run                         # shows error (app not in maintenance).
./bin/run redis:maintenance -a test_app_single_redis --run -f                      # shows maintenance result.
./bin/run redis:maintenance -a test_app_multiple_redis                             # shows error.
./bin/run redis:maintenance addon-name -a test_app_multiple_redis -w "Tue 23:00"   # sets the weekly maintenance window.
```

You can add Heroku Redis add-ons to any test app with `heroku addons:create heroku-redis -a test-app`.

## SOC2 Compliance
[GUS Work Item](https://gus.lightning.force.com/a07EE00001axJs6YAE)
